File: Persistencia de objetos/Ejercicio 3/Python/Main.py
from ArchivoCliente import ArchivoCliente
from Cliente import Cliente

def main():
    import os

    ruta = os.path.join(os.path.dirname(__file__), "clientes.txt")
    archivo = ArchivoCliente("C:/Users/teran/OneDrive/Desktop/Progra II/Practica 3 aux/PERSISTENCIA DE OBJETOS/Ejercicio 3/Python/clientes.txt")

    # Guardar clientes
    archivo.guardarCliente(Cliente(1, "Ana", 9374))
    archivo.guardarCliente(Cliente(2, "Luis", 9474))
    archivo.guardarCliente(Cliente(3, "Maria", 8493))

    c1 = archivo.buscarCliente(2)
    if c1:
        print("Encontrado por ID:", c1)
    else:
        print("Cliente no encontrado por ID")

    c2 = archivo.buscarCelularCliente(3333)
    if c2:
        print("Encontrado por teléfono:", c2)
    else:
        print("Cliente no encontrado por teléfono")

if __name__ == "__main__":
    main()

File: Persistencia de objetos/Ejercicio 3/Java/Main.java
import java.io.IOException;

public class Main {
    public static void main(String[] args) {
        ArchivoCliente archivo = new ArchivoCliente("clientes.txt");

        try {
            archivo.crearArchivo();

            archivo.guardarCliente(new Cliente(1, "Ana", 5936));
            archivo.guardarCliente(new Cliente(2, "Luis", 2398));
            archivo.guardarCliente(new Cliente(3, "Maria", 8484));


            Cliente c1 = archivo.buscarCliente(2);
            if (c1 != null) {
                System.out.println("Encontrado por ID: " + c1);
            } else {
                System.out.println("Cliente no encontrado por ID");
            }


            Cliente c2 = archivo.buscarCelularCliente(8484);
            if (c2 != null) {
                System.out.println("Cliente encontrado por teléfono: " + c2);
            } else {
                System.out.println("Cliente no encontrado por teléfono");
            }

        } catch (IOException e) {
            System.out.println("Error al trabajar con archivo: " + e.getMessage());
        }
    }
}